distutils/_msvccompiler.py

```diff
@@ -242,13 +268,12 @@ def initialize(self, plat_name=None):
         if plat_name is None:
             plat_name = get_platform()
         # sanity check for platforms to prevent obscure errors later.
-        if plat_name not in PLAT_TO_VCVARS:
+        if plat_name not in _vcvars_names:
             raise DistutilsPlatformError(
-                f"--plat-name must be one of {tuple(PLAT_TO_VCVARS)}"
+                f"--plat-name must be one of {tuple(_vcvars_names)}"
             )
 
-        # Get the vcvarsall.bat spec for the requested platform.
-        plat_spec = PLAT_TO_VCVARS[plat_name]
+        plat_spec = _get_vcvars_spec(get_host_platform(), get_platform())
```

Comment:
Seems the bug is in line 276 where it's ignoring the `plat_name` parameter and instead calling `get_platform()` again. 👀

Reply:
I think you're right, the following is enough of a change:
```suggestion
        plat_spec = _get_vcvars_spec(get_host_platform(), plat_name)
```